### PR TITLE
upgrade version + improved install script

### LINKED ISSUE #151: implement `gnmic version upgrade`
To upgrade to the latest available version

### PR DESCRIPTION
**Pull request by hellt**
1. added `gnmic version upgrade` command
2. removed hardcoded strings in the installation script and made `--use-pkg` flag that will be able to fetch the package and install it instead of using tar.gz archive

close #151

### CODE REVIEW

File: cmd/versionUpgrade.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	Use:   "upgrade",
+	Short: "upgrade gnmic to latest available version",
+
+	Run: func(cmd *cobra.Command, args []string) {
+		f, err := ioutil.TempFile("", "gnmic")
+		defer os.Remove(f.Name())
+		if err != nil {
+			log.Fatalf("Failed to create temp file %s\n", err)
+		}
+		downloadFile(downloadURL, f)
```
Comment (karimra) [inline cmd/versionUpgrade.go:27]: error is ignored here

File: cmd/versionUpgrade.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+		c := exec.Command("bash", f.Name())
+		c.Stdout = os.Stdout
+		c.Stderr = os.Stderr
+		err = c.Run()
+		if err != nil {
+			log.Fatalf("Upgrade failed: %s\n", err)
```
Comment (karimra) [inline cmd/versionUpgrade.go:34]: let's avoid log.Fatal and return an error

File: cmd/versionUpgrade.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+// upgradeCmd represents the version command
+var upgradeCmd = &cobra.Command{
+	Use:   "upgrade",
+	Short: "upgrade gnmic to latest available version",
+
+	Run: func(cmd *cobra.Command, args []string) {
+		f, err := ioutil.TempFile("", "gnmic")
+		defer os.Remove(f.Name())
+		if err != nil {
+			log.Fatalf("Failed to create temp file %s\n", err)
```
Comment (karimra) [inline cmd/versionUpgrade.go:25]: need to use logger, not log here. or return an error

**Review by karimra — CHANGES_REQUESTED** [review]

File: cmd/versionUpgrade.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+		if err != nil {
+			log.Fatalf("Upgrade failed: %s\n", err)
+		}
+	},
+}
+
+// downloadFile will download a file from a URL and write its content to a file
+func downloadFile(url string, file *os.File) error {
+	// Get the data
+	resp, err := http.Get(url)
```
Comment (karimra) [inline cmd/versionUpgrade.go:42]: better use a custom http client here, with a timeout (10s ?) .
The default http client doesn't have one

Comment (hellt) [reply cmd/versionUpgrade.go]: addressed in 7ec3633

File: cmd/versionUpgrade.go (hunk 3 above)
Comment (hellt) [reply cmd/versionUpgrade.go]: addressed in 7ec3633

File: cmd/versionUpgrade.go (hunk 2 above)
Comment (hellt) [reply cmd/versionUpgrade.go]: addressed in 7ec3633

File: cmd/versionUpgrade.go (hunk 1 above)
Comment (hellt) [reply cmd/versionUpgrade.go]: addressed in 7ec3633

**Comment by hellt** [comment]
will add docs

**Review by karimra — APPROVED** [review]

**Pull request merged**